ha/setup/post_disruptive_upgrade.py

```diff
@@ -0,0 +1,83 @@
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# abcheck_for_any_resource_presenceout this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+"""Module to do post-upgrade routines in scope of Disruptive Upgrade feature."""
+
+from cortx.utils.log import Log
+from ha.execute import SimpleCommand
+from ha.const import (BACKUP_DEST_DIR_CONF, CONFIG_DIR)
+from ha.setup.create_pacemaker_resources import create_all_resources
+
+
+def check_for_any_resource_presence() -> None:
+    '''Check if any resources are already present in a cluster.
+       if yes, means, pre-upgrade steps failes hence exit.'''
+
+    Log.info('Check for any resource presence in a cluster')
+
+    list_pcs_resources = 'crm_resource --list-raw'
+    resource_list = SimpleCommand().run_cmd(list_pcs_resources)
+
+    resource_list.split('\n')
+
+    if resource_list:
+         Log.error('Some resources are already present. \
```

Comment:
will performing the Post-Upgrade again will help resolve this issue?
If not, I think this error log should provide the correct next step.

Reply:
I think, there is no going back in the upgrade process. If something fails, we have to start the process from the scratch. So, if resources are present, means, pre-upgrade did not succeed. Means, we need to perform pre-upgrade again. Will update this in the message.